custom_components/ha_washdata/cycle_detector.py

```
@@ -494,6 +522,92 @@ def _sanitize_expected_duration(
+        for ts, power in reversed(self._power_readings):
+            if (timestamp - ts).total_seconds() > window_s:
+                break
+            window.append((ts, float(power)))
+        if len(window) < SMART_TERM_TAIL_MIN_POINTS:
+            return None
+        window.reverse()
+        span = (window[-1][0] - window[0][0]).total_seconds()
+        if span <= 0:
+            return None
+        energy = 0.0
+        for (t0, p0), (t1, p1) in zip(window, window[1:]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)